### ISSUE TITLE
Yandex ML handbook

### ISSUE THREAD

**Issue by aigoncharov** [aigoncharov/ml_misfits#2]

**Comment by aigoncharov** [comment]
Due to a popular demand, meetings have been move to Friday. See you on July 11!

**Comment by aigoncharov** [comment]
> # HW1
> 
> **Дедлайн:** 8 июля **Что делаем:** Читаем главы 1 (все подглавы) и 2.1. Делаем практические задания.
> 
> Ставьте 👍 если собираетесь подключиться!

Пулл-реквест с первой домашкой https://github.com/aigoncharov/ml_misfits/pull/10
Ревью и комментарии только приветствуются!

**Comment by aigoncharov** [comment]
# HW1

**Дедлайн:** 18 июля
**Что делаем:**
Читаем главу 2.1. Делаем практические задания.

Ставьте 👍 если собираетесь подключиться!

**Comment by aigoncharov** [comment]
В отчаянных попытках оптимизации сдвинул время созвона на 14:45

**Comment by aigoncharov** [comment]
# HW3

**Дедлайн:** 25 июля
**Что делаем:**
Читаем главу 2.2. Делаем практические задания.

Ставьте 👍 если собираетесь подключиться!

**Comment by aigoncharov** [comment]
По HW2 у кого-то сошлись оценки по кросс-валидации со значениями из контеста?
У меня вышло `np.float64(23324.8824602634), np.float64(21257.31418556281)`

Код:
```python
from sklearn.metrics import make_scorer
from sklearn.pipeline import Pipeline
from sklearn.model_selection import KFold
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import cross_val_score
import pandas as pd

data_preprocessing = [
    ("BaseDataPreprocessor", BaseDataPreprocessor(needed_columns=continuous_columns)),
]

pipe_linear = Pipeline(
    [
        *data_preprocessing,
        ("LinearRegression", LinearRegression()),
    ]
)

pipe_exponential = Pipeline(
    [
        *data_preprocessing,
        ("ExponentialLinearRegression", ExponentialLinearRegression()),
    ]
)

cv = KFold(n_splits=5, shuffle=True, random_state=42)

X_all = pd.concat([data_train, data_test], axis=0)
Y_all = np.concat([Y_train, Y_test], axis=0)

mae_scorer = make_scorer(mean_absolute_error)

linear_score = cross_val_score(pipe_linear, X_all, Y_all, cv=cv, scoring=mae_scorer)
exp_score = cross_val_score(pipe_exponential, X_all, Y_all, cv=cv, scoring=mae_scorer)

linear_score.mean(), exp_score.mean()
```

**Comment by aigoncharov** [comment]
Звонок передвинут на 17:00 (BST) по пятницам

**Comment by ruslansenatorov** [comment]
Рекомендую параллельно проходить мои курсы на степик.

<img width="990" height="790" alt="Image" src="https://github.com/user-attachments/assets/3c12e29c-06bc-43f5-9a4b-1a76a45c649a" />

Простая линейная регрессия (https://t.me/RuslanSenatorov/2526)
Множественная регрессия.Нормальное уравнение (https://t.me/RuslanSenatorov/2801)
Сингулярное разложение.Линейная регрессия (https://t.me/RuslanSenatorov/3088)
Градиентный спуск и Стохастический (https://t.me/RuslanSenatorov/2951)

**Comment by aigoncharov** [comment]
Иллюстрация почему Манхеттенска дистанция менее чувствительна к выбросам - https://colab.research.google.com/drive/1TJDTvFaFpe7G94RkqIKgZ9jjwlGUjfcN?usp=sharing

**Comment by aigoncharov** [comment]
Добавил доп лекции Воронцова http://www.machinelearning.ru/wiki/index.php?title=%D0%9C%D0%B0%D1%88%D0%B8%D0%BD%D0%BD%D0%BE%D0%B5_%D0%BE%D0%B1%D1%83%D1%87%D0%B5%D0%BD%D0%B8%D0%B5_%28%D0%BA%D1%83%D1%80%D1%81_%D0%BB%D0%B5%D0%BA%D1%86%D0%B8%D0%B9%2C_%D0%9A.%D0%92.%D0%92%D0%BE%D1%80%D0%BE%D0%BD%D1%86%D0%BE%D0%B2%29

**Comment by aigoncharov** [comment]
# HW3

Дедлайн: 1 августа.

Что делаем:
- Читаем "2.3. Решающие деревья";
- Прорешиваем TBD

**Comment by GTGAMESTUDIO** [comment]
Хороший сборник лекций, семинаров и практик от DLS 
https://stepik.org/org/dlschool

**Comment by GTGAMESTUDIO** [comment]
> > Хороший сборник лекций, семинаров и практик от DLS https://stepik.org/org/dlschool
> 
> Взял на пробу у них первый семестр за весну 2025 года из архива. Увы, там тоже нет домашки на деревья и метрические методы. Буду еще искать

Тоже курс от яндекса примерно параллелен [Yandex ML handbook](https://education.yandex.ru/handbook/ml), только с практикой.

**Comment by GTGAMESTUDIO** [comment]
[лекция Решающие деревья, композиции деревьев, Random Forest](https://youtube.com/live/rBIVch1h5qc)
[практика Решающие деревья, композиции деревьев, Random Forest](https://contest.yandex.ru/contest/53020/enter)

**Comment by GTGAMESTUDIO** [comment]
небольшой разбор
Степенной метод [википедия](https://www.google.com/url?q=https%3A%2F%2Fru.wikipedia.org%2Fwiki%2F%25D0%25A1%25D1%2582%25D0%25B5%25D0%25BF%25D0%25B5%25D0%25BD%25D0%25BD%25D0%25BE%25D0%25B9_%25D0%25BC%25D0%25B5%25D1%2582%25D0%25BE%25D0%25B4)
`def get_dominant_eigenvalue_and_eigenvector(A, num_steps):
    n = A.shape[0]
    v = np.random.rand(n) 
    v = v / np.linalg.norm(v) 

    for i in range(num_steps):
        v = A @ v  
        v = v / np.linalg.norm(v)  

    l = (v.T @ A @ v) / (v.T @ v)

    return float(l), v`

**Comment by aigoncharov** [comment]
# HW4

Дедлайн: 8 августа.

Что делаем:
- Читаем "2.4. Ансамбли в машинном обучении";
- Прорешиваем https://github.com/esokolov/ml-course-hse/tree/master/2024-fall/homework-practice/homework-practice-05-trees и https://github.com/esokolov/ml-course-hse/blob/master/2024-fall/homework-practice/homework-practice-06-boosting/homework_practice_06_boosting.ipynb

**Comment by GTGAMESTUDIO** [comment]
Хороший курс по ML от  К.В. Воронцова. 
Хетбук бывает на него ссылается
[Курс «Машинное обучение» 2019](https://www.youtube.com/playlist?list=PLJOzdkh8T5krxc4HsHbB8g8f0hu7973fK)

**Comment by aigoncharov** [comment]
Нашел интуитивное объяснение сигмоиды. Оказывается, это просто частны случай softmax!
https://stats.stackexchange.com/questions/233658/softmax-vs-sigmoid-function-in-logistic-classifier
https://willwolf.io/2017/04/19/deriving-the-softmax-from-first-principles/

**Comment by aigoncharov** [comment]
> Нашел интуитивное объяснение сигмоиды. Оказывается, это просто частны случай softmax! https://stats.stackexchange.com/questions/233658/softmax-vs-sigmoid-function-in-logistic-classifier https://willwolf.io/2017/04/19/deriving-the-softmax-from-first-principles/

У меня все интуиция не сходилась при переходе от логичного софтмакса, где мы просто нормализуем, к сигмоиде, где мы считали, что z - внезапно log(odds). Выпытал интуитивное [объяснение у чатгпт](https://chatgpt.com/share/6898cedc-91b4-8003-8aeb-905e20369713).

**Comment by GTGAMESTUDIO** [comment]
[ML Яндекс.pdf](https://github.com/user-attachments/files/21803665/ML.pdf)
Мини конспект по ml classic

**Comment by aigoncharov** [comment]
Воронцов в лекции рассказывает про вероятностный взгляд на реугляризацию - https://www.youtube.com/live/YaJ-QfSHl3o?si=lK1yNO9djsSnkwmc&t=5520
L2 норма форсит веса быть в нормальном распределении, а L1 - Лапласс.
Проперся и сделал визуализатор
https://colab.research.google.com/drive/1DyP9q8R_i6YADsQKBdNy_oul4D1Jq8v9?usp=sharing

**Comment by aigoncharov** [comment]
# HW6

Дедлайн: 5 сентября
Что делаем:
- Решаем https://github.com/girafe-ai/ml-course/blob/23f_yandex_ml_trainings/homeworks/assignment05_bagging_and_oob/assignment_bagging_and_oob.ipynb

**Comment by aigoncharov** [comment]
Залил домашку по bagging с векторизацией
https://github.com/aigoncharov/ml_misfits/blob/main/courses/yandex_ml_handbook/bagging/assignment_bagging_and_oob.ipynb

**Comment by aigoncharov** [comment]
# HW7

Дедлайн: 26 сентября.

Что делаем:
https://github.com/esokolov/ml-course-hse/blob/master/2024-fall/homework-practice/homework-practice-06-boosting/homework_practice_06_boosting.ipynb

**Внимание!** След звонок 26 сентября
